### PR TITLE
nixosTests.systemd: port to Python

### PR DESCRIPTION
**Pull request by andrew-d**
###### Motivation for this change

#72828 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @tfc @aszlig @flokli @abbradar

### CODE REVIEW

**Comment by mmahut** [comment]
@GrahamcOfBorg test systemd

File: nixos/tests/systemd.nix [hunk 1]
```diff
@@ -53,50 +53,66 @@ import ./make-test.nix ({ pkgs, ... }: {
...
+
+    with subtest("systemd-shutdown works"):
+        machine.shutdown()
+        machine.wait_for_unit("multi-user.target")
+        machine.succeed("test -e /tmp/shared/shutdown-test")
+
+    # Test settings from /etc/sysctl.d/50-default.conf are applied
+    with subtest("systemd sysctl settings are applied"):
+        machine.wait_for_unit("multi-user.target")
+        machine.succeed('sysctl net.core.default_qdisc | grep -q "fq_codel"')
```
Comment (flokli) [inline nixos/tests/systemd.nix]:
```suggestion
        assert "fq_codel" in machine.succeed('sysctl net.core.default_qdisc"')
```

File: nixos/tests/systemd.nix [hunk 2]
```diff
@@ -53,50 +53,66 @@ import ./make-test.nix ({ pkgs, ... }: {
...
-     $machine->succeed('systemctl show testservice1.service -p IOAccounting | grep -q "yes"');
-     $machine->succeed('systemctl status testservice1.service | grep -q "CPU:"');
-   };
+    # Test cgroup accounting is enabled
+    with subtest("systemd cgroup accounting is enabled"):
+        machine.wait_for_unit("multi-user.target")
+        machine.succeed(
+            'systemctl show testservice1.service -p IOAccounting | grep -q "yes"'
+        )
+        machine.succeed('systemctl status testservice1.service | grep -q "CPU:"')
```
Comment (flokli) [inline nixos/tests/systemd.nix]:
```suggestion
        assert "CPU:" in machine.systemctl('status testservice1.service"')
```

File: nixos/tests/systemd.nix (hunk 1 above)
Comment (andrew-d) [reply nixos/tests/systemd.nix]: :+1:

File: nixos/tests/systemd.nix [hunk 3]
```diff
@@ -53,50 +53,66 @@ import ./make-test.nix ({ pkgs, ... }: {
...
+                "vm-state-machine/empty0.qcow2",
+                "x-initrd-mount.raw",
+            ]
+        )
+        extinfo = subprocess.check_output(
+            [
+                "${pkgs.e2fsprogs}/bin/dumpe2fs",
+                "x-initrd-mount.raw",
+            ]
+        ).decode("latin-1")
```
[the remark this answers is not in the document]
Comment (andrew-d) [reply nixos/tests/systemd.nix]: Looks like it :+1:

File: nixos/tests/systemd.nix (hunk 2 above)
Comment (andrew-d) [reply nixos/tests/systemd.nix]: :+1:

**Comment by andrew-d** [comment]
@flokli - Thanks for the review - made those changes, along with a few more cleanups.

**Comment by andrew-d** [comment]
@flokli - And a ping on this one as well!

**Review by tfc — APPROVED** [review]

**Comment by tfc** [comment]
cc @worldofpeace

**Comment by andrew-d** [comment]
@GrahamcOfBorg build nixosTests.systemd

**Comment by worldofpeace** [comment]
Merging as is, because it was broken before https://hydra.nixos.org/job/nixos/trunk-combined/nixos.tests.systemd.x86_64-linux

**Pull request merged**